source/GameInterface.Tests/Serialization/SerializerTests/WorkshopSerializationTest.cs

```diff
@@ -0,0 +1,116 @@
+﻿using GameInterface.Serialization;
+using GameInterface.Serialization.Impl;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.CampaignSystem.Settlements.Workshops;
+using TaleWorlds.Core;
+using TaleWorlds.Library;
+using TaleWorlds.Localization;
+using TaleWorlds.ObjectSystem;
+using Xunit;
+
+namespace GameInterface.Tests.Serialization.SerializerTests
+{
+    public class WorkshopSerializationTest
+    {
+        public WorkshopSerializationTest()
+        {
+            MBObjectManager.Init();
+            MBObjectManager.Instance.RegisterType<Settlement>("Settlement", "Settlements", 4U, true, false);
+            MBObjectManager.Instance.RegisterType<Hero>("Hero", "Heros", 5U, true, false);
+            MBObjectManager.Instance.RegisterType<MobileParty>("MobileParty", "MobilePartys", 5U, true, false);
```

Comment:
Would these typeIds and naming being different from base game be wrong later on? Not sure what they actually do, for example MobileParty is registered as ("MobileParty", "MobileParties", 14u, true, true).

Reply:
Naming doesn't matter here, the only thing that matters is the `<Type>`